### PR TITLE
feat: 🎸 Support fetching historical instructions for a DID

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

Fetch historical instructions using middleware v2 where a DID is involved

### Breaking Changes

NA

### JIRA Link

DA-168

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

**Comment by polymath-eric** [comment]
There is a smell:

`status: status as InstructionStatusEnum,` could be just `status` I think

I tried locally removing the assertion and things still built. guessing its in the base branch here.

File: src/middleware/queriesV2.ts [hunk 1]
```diff
@@ -319,6 +319,70 @@ export function instructionsQuery(
...
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  return {
+    query,
+    variables: undefined,
```
Comment (polymath-eric) [inline src/middleware/queriesV2.ts:382]: (nit) Should `identityId` be a GraphQL variable? 

Looks like its getting string interpolated which doesn't seem as nice as the first class variable to me.

That said, if it needs to be like that for some reason or is more consistent its OK to leave

**Review by polymath-eric — APPROVED** [review]
LGTM, 

Left one nitty comment

**Review by polymath-eric — APPROVED** [review]

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**